k8_cortx_cloud/upgrade-cortx-cloud.sh

```diff
@@ -270,6 +282,77 @@ function initiate_upgrade() {
     done <<< "$(kubectl get deployments --output=jsonpath="{range .items[*]}{.metadata.name}{'\n'}{end}" --namespace="${NAMESPACE}" | grep "$1")"
 }
 
+function prepare_upgrade() {
+    POD_TYPE=$1
+    if [[ "$POD_TYPE" == "cortx" ]] || [[ "$POD_TYPE" == "control" ]]; then
+        control_image=$(parse_solution 'solution.images.cortxcontrol' | cut -f2 -d'>')
+        RULES=""
+        get_compatibility_clauses "$control_image"
+        control_nodes=("cortx-control")
+        check_version_compatibility $control_nodes
+    fi
+    if [[ "$POD_TYPE" == "cortx" ]] || [[ "$POD_TYPE" == "data" ]]; then
+        data_image=$(parse_solution 'solution.images.cortxdata' | cut -f2 -d'>')
+        RULES=""
+        get_compatibility_clauses "$data_image"
+        data_nodes=$(kubectl get svc -n "${NAMESPACE}" | grep cortx-data-headless* | awk '{print $1}')
+        check_version_compatibility $data_nodes
+    fi
+    if [[ "$POD_TYPE" == "cortx" ]] || [[ "$POD_TYPE" == "server" ]]; then
+        server_image=$(parse_solution 'solution.images.cortxserver' | cut -f2 -d'>')
+        RULES=""
+        get_compatibility_clauses "$server_image"
+        server_nodes=$(kubectl get svc -n "${NAMESPACE}" | grep cortx-server-headless* | awk '{print $1}')
+        check_version_compatibility $server_nodes
+    fi
+    if [[ "$POD_TYPE" == "cortx" ]] || [[ "$POD_TYPE" == "ha" ]]; then
+        ha_image=$(parse_solution 'solution.images.cortxha' | cut -f2 -d'>')
+        RULES=""
+        get_compatibility_clauses "$ha_image"
+        ha_nodes=$(kubectl get svc -n "${NAMESPACE}" | grep cortx-ha-headless* | awk '{print $1}')
+        check_version_compatibility $ha_nodes
+    fi
+}
+
+function get_compatibility_clauses() {
+    REQUIRES=$(docker inspect --format='{{ index .Config.Labels "org.opencontainers.image.version"}}' $1)
+    IFS=',' read -ra  newarr <<< "$REQUIRES"
+    for val in "${newarr[@]}";
+    do
+      val=${val//[[:blank:]]/}
+      RULES+='"'$val'",'
+    done
+    RULES="{\"requires\":[${RULES%?}]}"
+}
+
+function check_version_compatibility() {
+  echo -e "\n--------------------------------------"
+  while IFS= read -r line; do
+    echo -e "Checking Version Compatibility for $line\n"
+    HOSTNAME=$(hostname)
+    PORT="31169"
+    endpoint="https://${HOSTNAME}:${PORT}/api/v2/version/compatibility/node/${line}"
+    response=$(curl -k -XPOST $endpoint -d ${RULES} -s | jq) 
+    echo "RESPONSE: $response"
```

Comment:
do we need to show complete response the the user?

Reply:
done removed the logging